### ISSUE TITLE
Gradle. `inputDir` tricky behaviour

### ISSUE THREAD

**Issue by jonnyzzz** [JetBrains/kotlin-native#775]
I have a hello-world like application. Sources under Gradle-standard location: `src/main/kotlin`. Anything works like a charm. 
Added an additional directory via 
```konanArtifacts {
  kpm {
    inputDir "$genMain"
  }
}```

Application no longer runs. 

It turned out, adding `inputDir` makes default path `src/main/kotlin` disappear. Let's make defaults behaviour more predictable

**Comment by ilmat192** [comment]
It's a correct behaviour. If we didn't remove the default path we would get an error for such a project:

```
konanArtifacts {
    lib {
        inputFiles fileTree('src/lib/kotlin')
        produce library
    }

    main {
        // Use the default source path and link with `lib`.
        library konanArtifacts['lib'].compilationTask.artifactPath
    }
}

```

Here both `lib` and `main` artifacts contain code from the default source path and we get a 'duplicate symbol' error at the link stage.
Looks like such an issue also may be solved by reworking the plugin according to native software model and using source sets.

**Comment by ilmat192** [comment]
I've created an issue for this work: https://github.com/JetBrains/kotlin-native/issues/778
If you have no objections I'll close this issue as a duplicate of #778.

**Comment by ilmat192** [comment]
Closed as a part of #778.
